registry/repository.go

```diff
@@ -21,176 +20,123 @@ const (
+	if err != nil {
+		return nil, ocispec.Descriptor{}, err
+	}
+	return sigBlob, sigDesc, nil
 }
 
-// PutSignatureManifest creates and uploads an signature artifact linking the manifest and the signature
-func (c *RepositoryClient) PutSignatureManifest(ctx context.Context, signature []byte, signatureMediaType string, subjectManifest notation.Descriptor, annotations map[string]string) (notation.Descriptor, SignatureManifest, error) {
-	signatureDesc, err := c.uploadSignature(ctx, signature, signatureMediaType)
+// PushSignature creates and uploads an signature manifest along with its
+// linked signature envelope blob. Upon successful, PushSignature returns
+// signature envelope blob and manifest descriptors.
+func (c *repositoryClient) PushSignature(ctx context.Context, blob []byte, mediaType string, subject ocispec.Descriptor, annotations map[string]string) (blobDesc, manifestDesc ocispec.Descriptor, err error) {
+	blobDesc, err = c.uploadSignature(ctx, blob, mediaType)
 	if err != nil {
-		return notation.Descriptor{}, SignatureManifest{}, err
+		return ocispec.Descriptor{}, ocispec.Descriptor{}, err
 	}
 
-	manifestDesc, err := c.uploadSignatureManifest(ctx, artifactDescriptorFromNotation(subjectManifest), signatureDesc, annotations)
+	manifestDesc, err = c.uploadSignatureManifest(ctx, subject, blobDesc, annotations)
 	if err != nil {
-		return notation.Descriptor{}, SignatureManifest{}, err
+		return ocispec.Descriptor{}, ocispec.Descriptor{}, err
 	}
 
-	signatureManifest := SignatureManifest{
-		Blob:        notationDescriptorFromArtifact(signatureDesc),
-		Annotations: annotations,
-	}
-	return notationDescriptorFromOCI(manifestDesc), signatureManifest, nil
+	return blobDesc, manifestDesc, nil
 }
 
-func (c *RepositoryClient) getArtifactManifest(ctx context.Context, manifestDigest digest.Digest) (artifactspec.Manifest, error) {
+// getSignatureManifest returns signature manifest given signature manifest
+// descriptor
+func (c *repositoryClient) getSignatureManifest(ctx context.Context, sigManifestDesc ocispec.Descriptor) (*artifactspec.Manifest, error) {
+
 	repo := c.Repository
 	repo.ManifestMediaTypes = []string{
 		artifactspec.MediaTypeArtifactManifest,
 	}
 	store := repo.Manifests()
-	desc, err := store.Resolve(ctx, manifestDigest.String())
-	if err != nil {
-		return artifactspec.Manifest{}, err
+	if sigManifestDesc.Size > maxManifestSizeLimit {
+		return &artifactspec.Manifest{}, fmt.Errorf("manifest too large: %d", sigManifestDesc.Size)
 	}
-	if desc.Size > maxManifestSizeLimit {
-		return artifactspec.Manifest{}, fmt.Errorf("manifest too large: %d", desc.Size)
-	}
-	manifestJSON, err := content.FetchAll(ctx, store, desc)
+	manifestJSON, err := content.FetchAll(ctx, store, sigManifestDesc)
 	if err != nil {
-		return artifactspec.Manifest{}, err
+		return &artifactspec.Manifest{}, err
 	}
 
-	var manifest artifactspec.Manifest
-	err = json.Unmarshal(manifestJSON, &manifest)
+	var sigManifest artifactspec.Manifest
```

Comment:
After fetching, the byte array should be unmarshaled to [ocispec.Artifact](https://github.com/opencontainers/image-spec/blob/v1.1.0-rc2/specs-go/v1/artifact.go#L19) so that the `artifactspec` can be deprecated.